tests/functional/test_add_session.py

```diff
@@ -0,0 +1,255 @@
+        db_func.add_session(1, "1/1/2020")
+        db_func.add_user("test", "test")
+        val_result, error, session_object = session_input_validator(
+            session_id, date, host
+        )
+        assert val_result == result
+        if val_result:
+            assert session_object.id == int(session_id)
+            assert datetime.datetime.strptime(
+                session_object.date, "%m/%d/%Y"
+            ) == datetime.datetime.strptime(date, "%Y-%m-%d")
+
+
+@pytest.mark.parametrize(
+    (
+        "beer_name",
+        "beer_abv",
+        "brewery",
+        "style",
+        "votes",
+        "win",
+        "username",
+        "session_id",
+        "result",
+    ),
+    (
+        (
+            "Good Beer",
+            "2.1",
+            "Good Brewery",
+            "IPA",
+            "1",
+            "1",
+            "test",
+            "1",
+            True,
+        ),  # Beer Name Cases
+        ("", "2.1", "Good Brewery", "IPA", "1", "1", "test", "1", False),
+        (
+            "Good Beer",
+            "2",
+            "Good Brewery",
+            "IPA",
+            "1",
+            "1",
+            "test",
+            "1",
+            True,
+        ),  # ABV Cases
+        ("Good Beer", "", "Good Brewery", "IPA", "1", "1", "test", "1", False),
+        ("Good Beer", "-1", "Good Brewery", "IPA", "1", "1", "test", "1", False),
+        ("Good Beer", "95.424", "Good Brewery", "IPA", "1", "1", "test", "1", True),
+        ("Good Beer", "string", "Good Brewery", "IPA", "1", "1", "test", "1", False),
+        ("Good Beer", "2.1", "", "IPA", "1", "1", "test", "1", False),  # Brewery Cases
+        (
+            "Good Beer",
+            "2.1",
+            "Good Brewery",
+            "",
+            "",
+            "1",
+            "test",
+            "1",
+            False,
+        ),  # Style Cases
+        (
+            "Good Beer",
+            "2.1",
+            "Good Brewery",
+            "IPA",
+            "-1",
+            "1",
+            "test",
+            "1",
+            False,
+        ),  # Vote Cases
+        ("Good Beer", "2.1", "Good Brewery", "IPA", "1.01", "1", "test", "1", False),
+        ("Good Beer", "2.1", "Good Brewery", "IPA", "41", "1", "test", "1", True),
+        (
+            "Good Beer",
+            "2.1",
+            "Good Brewery",
+            "IPA",
+            "1",
+            "0",
+            "test",
+            "1",
+            True,
+        ),  # Win Cases
+        ("Good Beer", "2.1", "Good Brewery", "IPA", "1", "-1", "test", "1", False),
+        ("Good Beer", "2.1", "Good Brewery", "IPA", "1", "2", "test", "1", False),
+        (
+            "Good Beer",
+            "2.1",
+            "Good Brewery",
+            "IPA",
+            "1",
+            "1",
+            "notarealuser",
+            "1",
+            False,
+        ),  # User cases
+    ),
+)
+def test_beer_input_validator(
```

Comment:
These tests should have argstrings.  Also I've found writing tests with the [arrange, act, assert](http://wiki.c2.com/?ArrangeActAssert) pattern in mind really helpful.  Specifically labeling these are really helpful too, e.g.:
```
test_beer_input_validator:
  # Arrange
   with app.app_context()
       # Act
       foobar = beer_input_validator
       
      # Assert
      assert val_restult == result
```

Reply:
We're lucky we have test code 😆 but yes I agree, I'll document. 

I've been following arrange, act, assert for the most part. Or well arrange, act, assert, act, assert ... as I still need to learn how to fully use pytest parameterize.